internal/controller/utils/nim.go

```diff
@@ -133,6 +134,37 @@ func GetAvailableNimRuntimes(logger logr.Logger) ([]NimRuntime, error) {
 
 // ValidateApiKey is used for validating the given API key by retrieving the token and pulling the given custom runtime
 func ValidateApiKey(logger logr.Logger, apiKey string, runtimes []NimRuntime) error {
+	if isPersonalApiKey(apiKey) {
+		return validatePersonalApiKey(logger, apiKey)
+	}
+	return validateLegacyApiKey(logger, apiKey, runtimes)
+}
+
+// validatePersonalApiKey is used for validating the given API key by retrieving the token and pulling the given custom runtime
+func validatePersonalApiKey(logger logr.Logger, apiKey string) error {
+	req, reqErr := http.NewRequest("POST", nimGetCallerInfoFmt, strings.NewReader(fmt.Sprintf("credentials=%s", apiKey)))
```

Comment:
nit: Typically, I'd suggest the following:

```suggestion
	req, reqErr := http.NewRequest("POST", nimGetCallerInfoFmt, strings.NewReader(fmt.Sprintf("credentials=%s", url.QueryEscape(apiKey))))
```

...but I guess in this case the apiKey won't contain any characters that need to be escaped, right?